File: 1538-maximum-points-you-can-obtain-from-cards/maximum-points-you-can-obtain-from-cards.cpp
class Solution {
public:

    int maxScore(vector<int>& cardPoints, int k) {
        int n = cardPoints.size();
        if(k==n){
            return accumulate(cardPoints.begin(), cardPoints.end(), 0);
        }
        int size = n-k;
        int total = accumulate(cardPoints.begin(), cardPoints.end(), 0);
        int sum = 0;
        for(int i = 0; i<size; i++){
            sum+=cardPoints[i];
        }
        int minSum = sum;
        int start = 0;
        for(int end = size; end<n; end++){
            sum+=cardPoints[end];
            sum-=cardPoints[start];
            start++;
            minSum = min(minSum, sum);
        }
        return total - minSum;
    }
};

File: 1538-maximum-points-you-can-obtain-from-cards/maximum-points-you-can-obtain-from-cards.py
class Solution:
    def maxScore(self, nums: List[int], k: int) -> int:
        n = len(nums)
        total = sum(nums)
        if k==n:
            return total
        size = n-k
        windSum = 0
        for i in range(size):
            windSum += nums[i]
        minSum = windSum
        start = 0
        for end in range(size, n):
            windSum+=nums[end]
            windSum-=nums[start]
            start+=1
            minSum = min(minSum, windSum)
        return total - minSum